### PR TITLE
boilerplate code added

### PR DESCRIPTION
**Pull request by geekanamika**
Boilerplate code for MVP design pattern
Design doc(as suggested) : https://docs.google.com/document/d/1tvuckcfnPufGBVpbZC0wizlSwg203O2n5i6tDKw9KfY/edit

### CODE REVIEW

**Comment by geekanamika** [comment]
@niksj @manjumegh Please refer to the design provided in PR, codacy errors with unused variables/empty method body can be ignored for now.

File: malaria-app-android/build.gradle [hunk 1]
```diff
@@ -26,9 +26,13 @@ android {
...
-    compile "com.android.support:support-v4:$rootProject.ext.supportLibraryVersion" // v4
-    compile "com.viewpagerindicator:library:$rootProject.ext.viewPagerIndicatorVersion"
-    compile files('libs/volleylib.jar')
-    compile files('libs/GraphView-3.1.2.jar')
+    implementation "com.android.support:support-v4:$rootProject.ext.supportLibraryVersion" // v4
+    implementation "com.viewpagerindicator:library:$rootProject.ext.viewPagerIndicatorVersion"
+    implementation "com.android.support:appcompat-v7:$rootProject.ext.supportLibraryVersion"
+    implementation files('libs/volleylib.jar')
+    implementation files('libs/GraphView-3.1.2.jar')
     testCompile 'junit:junit:4.12'
```
Comment (m-murad) [inline malaria-app-android/build.gradle]: Update this configuration. `testCompile -> testImplementation`

**Comment by geekanamika** [comment]
@m-murad updated file as suggested, Can you final review & approve if it's okay?

File: malaria-app-android/src/main/java/com/peacecorps/malaria/ui/userprofile/UserProfileActivity.java [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+                    // user = user.getAppUser(name,email, Integer.parseInt(age),userMedicineType);
+                   // postUserDetails(user);
+                    setNewDetails();
+                }
+
+            }
+        };
+    }
+
+    // TODO: remove comments & fix the error after community discussion, ;)
```
Comment (niksj) [inline malaria-app-android/src/main/java/com/peacecorps/malaria/ui/userprofile/UserProfileActivity.java:171]: Why is this code commented?

Comment (niksj) [reply malaria-app-android/src/main/java/com/peacecorps/malaria/ui/userprofile/UserProfileActivity.java:171]: Okay, got it. Just checked the discussion on issue #281

**Review by niksj — APPROVED** [review]
Looks good to me.

File: malaria-app-android/src/main/java/com/peacecorps/malaria/ui/base/BaseFragment.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+     * Layout resource to inflate
+     *
+     * @return layout resource
+     */
+    @LayoutRes
+    protected abstract int getContentResource();
+
+    protected abstract void init(View view, @Nullable Bundle savedInstanceState);
+
+    void displayToastMessage(String message) {
```
[the remark this answers is not in the document]
Comment (niksj) [reply malaria-app-android/src/main/java/com/peacecorps/malaria/ui/base/BaseFragment.java:56]: @geekanamika It states to mention scope of the method: displayToastMessage(public/private) rather that keeping it default which is package private level.

**Review by manjumegh — APPROVED** [review]
Looks good to me, I hope few codacy error will be taken care, once you fill in all the boiler plate code.

**Comment by r00pal** [comment]
@geekanamika  can you also close the issues that this PR fixes?

**Pull request merged**